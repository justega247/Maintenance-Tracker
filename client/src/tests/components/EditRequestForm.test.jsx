import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedEditRequest, { EditRequestForm } from '../../components/EditRequestForm';
import mockData from '../__mocks__/mockData';

let wrapper;
let props;

const createMockStore = configureMockStore([thunk]);
const store = createMockStore({});

describe('Edit Request Form', () => {
  beforeEach(() => {
    props = {
      fetchRequest: jest.fn().mockResolvedValue(Promise.resolve()),
      startUpdateRequest: jest.fn(),
      history: {
        push: jest.fn(),
      },
      match: {
        params: {
          id: 2,
        },
      },
    };
    wrapper = shallow(<EditRequestForm {...props} />);
  });

  test('should render EditRequestForm correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should set title on input change', () => {
    const value = 'broken table';
    const name = 'title';
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
    });
    expect(wrapper.state(name)).toBe(value);
  });

  test('should call the submit props', () => {
    const { request } = mockData;
    const data = {
      ...request,
      title: '',
      description: '',
    };
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState(data);
    const instance = wrapper.instance();
    instance.onSubmit(event);
    expect(wrapper).toMatchSnapshot();
  });

  test('should call startUpdateRequest with valid submission', () => {
    const { request } = mockData;
    wrapper.setState(request);
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {},
    });
    expect(props.startUpdateRequest).toHaveBeenCalledWith(2, request, props.history);
  });

  test('should render connected component', () => {
    wrapper = shallow(<ConnectedEditRequest store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
