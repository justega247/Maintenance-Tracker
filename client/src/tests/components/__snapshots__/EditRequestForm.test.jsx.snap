// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Request Form should call the submit props 1`] = `
<div
  className="create__request"
>
  <div>
    <div>
      <h1
        className="create__request-title"
      >
        Edit Request
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      className="request__form"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            className="label request__label"
            htmlFor="type"
          >
            Type:
            <select
              className="request__type"
              name="type"
              onChange={[Function]}
              value="repairs"
            >
              <option
                key="repairs"
                value="repairs"
              >
                repairs
              </option>
              <option
                key="maintenance"
                value="maintenance"
              >
                maintenance
              </option>
            </select>
          </label>
        </div>
        <div>
          <label
            className="label request__label"
            htmlFor="title"
          >
            Title:
            <div>
              <input
                className="form__input"
                name="title"
                onChange={[Function]}
                placeholder="enter a title"
                type="text"
                value=""
              />
            </div>
          </label>
          <div
            className="form__error"
            key="Sorry, you have to specify a title for your request"
          >
            Sorry, you have to specify a title for your request
          </div>
        </div>
        <div>
          <label
            className="label request__label"
            htmlFor="description"
          >
            Description:
            <div>
              <textarea
                className="form__input"
                name="description"
                onChange={[Function]}
                placeholder="enter a description"
                value=""
              />
            </div>
          </label>
          <div
            className="form__error"
            key="Sorry, you have to specify a description for your request"
          >
            Sorry, you have to specify a description for your request
          </div>
        </div>
        <div>
          <Button
            className="button--big"
            text="Edit Request"
            type="submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Edit Request Form should render EditRequestForm correctly 1`] = `
<div
  className="create__request"
>
  <div>
    <div>
      <h1
        className="create__request-title"
      >
        Edit Request
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      className="request__form"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            className="label request__label"
            htmlFor="type"
          >
            Type:
            <select
              className="request__type"
              name="type"
              onChange={[Function]}
              value=""
            >
              <option
                key="repairs"
                value="repairs"
              >
                repairs
              </option>
              <option
                key="maintenance"
                value="maintenance"
              >
                maintenance
              </option>
            </select>
          </label>
        </div>
        <div>
          <label
            className="label request__label"
            htmlFor="title"
          >
            Title:
            <div>
              <input
                className="form__input"
                name="title"
                onChange={[Function]}
                placeholder="enter a title"
                type="text"
                value=""
              />
            </div>
          </label>
        </div>
        <div>
          <label
            className="label request__label"
            htmlFor="description"
          >
            Description:
            <div>
              <textarea
                className="form__input"
                name="description"
                onChange={[Function]}
                placeholder="enter a description"
                value=""
              />
            </div>
          </label>
        </div>
        <div>
          <Button
            className="button--big"
            text="Edit Request"
            type="submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Edit Request Form should render connected component 1`] = `
<EditRequestForm
  fetchRequest={[Function]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 2,
      },
    }
  }
  startUpdateRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
